lib/Sema/CSSimplify.cpp

```diff
@@ -2196,6 +2231,33 @@ bool ConstraintSystem::repairFailures(
     break;
   }
 
+  case ConstraintLocator::ApplyArgToParam: {
+    auto loc = getConstraintLocator(locator);
+    if (repairByInsertingExplicitCall(lhs, rhs))
+      return true;
+
+    if (lhs->getOptionalObjectType() && !rhs->getOptionalObjectType()) {
+      conversionsOrFixes.push_back(
+          ForceOptional::create(*this, lhs, lhs->getOptionalObjectType(), loc));
+    }
+
+    auto anchor = simplifyLocatorToAnchor(*this, loc);
+
+    if (auto *fix = fixPropertyWrapperFailure(
```

Comment:
Instead of duplicating logic common to `LValueConversion` and `ApplyArgToParam` maybe it would make sense to just do `if (elt.getKind() != ConstraintLocator::ApplyArgToParam) break;` right before trying to fix property wrapper related errors?